cmd/juju/commands/bootstrap.go

```diff
@@ -653,6 +651,15 @@ to create a new model to deploy k8s workloads.
 		}
 	}
 
+	cfgControllerName, _ := bootstrapCfg.controller[controller.ControllerName].(string)
+	if c.controllerName != "" && cfgControllerName != "" {
+		return errors.NotValidf("controller name and controller config name both set, only one allowed")
```

Comment:
this will print 
"controller name and controller config name both set, only one allowed not valid"

you need to use the errors.NewNotValid() form

also there's no unit tests for this code (would have picked up the dodgy error message)

Reply:
Done & Done